sklearn/model_selection/tests/test_validation.py

```diff
@@ -1299,3 +1302,19 @@ def test_permutation_test_score_pandas():
         check_series = lambda x: isinstance(x, TargetType)
         clf = CheckingClassifier(check_X=check_df, check_y=check_series)
         permutation_test_score(clf, X_df, y_ser)
+
+
+def test_cross_val_score_labels_paramter():
+    # check if adding a classes argument to estimator passes it to the scorer
+    X, y = make_blobs(n_samples=90, n_features=2, centers=3, shuffle=False)
+    clf1 = GaussianNB()
+    clf2 = GaussianNB(classes=[0, 1, 2], priors=[0.33, 0.33, 0.34])
+
+    scores1 = cross_val_score(clf1, X, y,
+                              cv=KFold(n_splits=2, shuffle=False),
+                              scoring='f1_macro')
```

Comment:
i'm sorry i didn't get your point. this is by design because micro score just aggregates all classes so the result should be the same for both cases. macro result should be greater in case 1 because in second case, averaging is done on 3 classes not 2.